layers.go

```diff
@@ -527,7 +527,11 @@ func (r *layerStore) SetMetadata(id, metadata string) error {
 }
 
 func (r *layerStore) tspath(id string) string {
-	return filepath.Join(r.layerdir, id+tarSplitSuffix)
+	tmpPath := ""
+	if DefaultStoreOptions.GraphRoot == "" {
+		tmpPath = filepath.Join(DefaultStoreOptions.GraphRoot, "tmp")
+	}
+	return filepath.Join(tmpPath, r.layerdir, id+tarSplitSuffix)
 }
```

Comment:
That file should only be written when we're applying a diff to a initialize a layer, though.  Were you attempting to create layers in a read-only location?